tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1462,3 +1462,257 @@ def test_metric_computation_handles_absent_labels():
+    mean_squared_error_data1_original = sklmetrics.mean_squared_error(eval1_y, pred1_y)
+    assert np.isclose(mean_squared_error_data1_original, mean_squared_error_data1)
+
+    lor_score_data1_original = model.score(eval1_X, eval1_y)
+    assert np.isclose(lor_score_data1_original, lor_score_data1)
+
+    pred1_y_original = model.predict(eval1_X)
+    assert np.allclose(pred1_y_original, pred1_y)
+
+
+def test_run_all_metric_examples():
+    import doctest
+    from sklearn import metrics
+    mlflow.sklearn.autolog()
+    for metric_name in mlflow.sklearn._get_metric_name_list():
+        metric_api = getattr(metrics, metric_name)
+        doctest.run_docstring_examples(metric_api.__doc__, {}, verbose=True)
+
+
+def test_post_training_post_training_metric_autologging_for_predict_prob():
+    import sklearn.linear_model
+
+    mlflow.sklearn.autolog()
+    from sklearn.metrics import roc_auc_score
+
+    X, y = get_iris()
+    lor_model = sklearn.linear_model.LogisticRegression(
+        solver="saga", max_iter=100, random_state=0
+    )
+    with mlflow.start_run() as run:
+        lor_model.fit(X, y)
+        y_prob = lor_model.predict_proba(X)
+        y_true_onehot = np.eye(3)[y]
+        roc_auc_metric = roc_auc_score(y_true_onehot, y_prob)
+
+    params, metrics, tags, artifacts = get_run_data(run.info.run_id)
+    assert metrics['roc_auc_score_X'] == roc_auc_metric
+
+
+def test_is_metrics_value_loggable():
+    is_metrics_value_loggable = \
+        mlflow.sklearn._autologging_metrics_manager.is_metrics_value_loggable
+    assert is_metrics_value_loggable(3)
+    assert is_metrics_value_loggable(3.5)
+    assert is_metrics_value_loggable(np.int(3))
+    assert is_metrics_value_loggable(np.float32(3.5))
+    assert not is_metrics_value_loggable(True)
+    assert not is_metrics_value_loggable(np.bool(True))
+    assert not is_metrics_value_loggable([1, 2])
+    assert not is_metrics_value_loggable(np.array([1, 2]))
+
+
+def test_nested_metric_call_is_disabled():
+    import sklearn.linear_model
+    mlflow.sklearn.autolog()
+
+    X, y = get_iris()
+    eval1_X, eval1_y = X[0::3], y[0::3]
+    lr_model = sklearn.linear_model.LinearRegression()
+
+    with mlflow.start_run():
+        # test post training metric logging disabled in fit scope
+        with mock.patch('mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric') \
+                as patched_log_post_training_metric:
+            lr_model.fit(X, y)
+            patched_log_post_training_metric.assert_not_called()
+
+        # test post training metric logging called only once in model.score
+        with mock.patch('mlflow.sklearn._AutologgingMetricsManager.log_post_training_metric') \
+                as patched_log_post_training_metric:
+            lr_model.score(eval1_X, eval1_y)
+            patched_log_post_training_metric.call_count == 1 and \
+                patched_log_post_training_metric.call_args[0][1] == 'r2_score_eval1_X'
```

Comment:
Are we missing `assert` calls here?

Reply:
```suggestion
            assert patched_log_post_training_metric.call_count == 1
            assert patched_log_post_training_metric.call_args[0][1] == 'r2_score_eval1_X'
```

Can we split the check here?